### PR TITLE
[Feature] New APP Request S3 Hookup

### PR DESCRIPTION
**Pull request by EmilioMena**
## Notion ticket link

[New APP Request S3 Hookup](https://www.notion.so/uwblueprintexecs/ac7b1f636a214a0cbf45f6b76f899b54?v=5414ee5fc5974f149921e192d6eb29b8)

## Implementation description
* Added next-s3-upload package
* Used next-s3-upload hook to upload pdf to s3

## Notes
* Validation of upload done before form validation
* Files will be stored in `next-s3-upload/{random_uuid}/{filename}` directory on bucket

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: pages/admin/request/create-new.tsx [hunk 1]
```diff
@@ -248,6 +252,23 @@ export default function CreateNew() {
...
+      try {
+        const { url } = await uploadToS3(guardianPOAFile);
+        setGuardianInformation({
+          ...guardianInformation,
+          poaFormUrl: url,
+        });
+      } catch (err) {
+        toast({
+          status: 'error',
+          description: `Failed to upload POA file: ${err}`,
```
Comment (EmilioMena) [inline pages/admin/request/create-new.tsx:265]: is this error message enough for now?

File: pages/api/s3-upload.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { APIRoute as default } from 'next-s3-upload';
```
Comment (EmilioMena) [inline pages/api/s3-upload.ts:1]: folder structure in s3 can be changed in this file if needed see 
https://next-s3-upload.codingvalue.com/s3-file-paths

File: pages/admin/request/create-new.tsx (hunk 1 above)
Comment (OustanDing) [reply pages/admin/request/create-new.tsx:265]: Yes, this works for now

**Review by OustanDing — COMMENTED** [review]
Is the POA form URL change being reflected in the DB as well?

File: pages/api/s3-upload.ts (hunk 2 above)
Comment (OustanDing) [reply pages/api/s3-upload.ts:1]: This seems like it will be useful when we add the other document types.

**Comment by EmilioMena** [comment]
> Is the POA form URL change being reflected in the DB as well?

The url will need to be signed when it is displayed to the user (i.e when fetching the url on the frontend) as it will only be valid for a certain amount of time. I don't think that's in the scope of this PR though

> Also, if there were changes that were made to .env, please add those additional fields to .env.sample too.

Good call will add that in!

File: pages/api/s3-upload.ts (hunk 2 above)
Comment (EmilioMena) [reply pages/api/s3-upload.ts:1]: +1 I'll add that as a comment in here so it's more visible

**Comment by EmilioMena** [comment]
poaFormObjectKey change in https://github.com/uwblueprint/richmond-centre-for-disability/pull/146/commits/391f5e7e050ec3b7f00142831ab437d91eadc529

**Review by OustanDing — APPROVED** [review]

**Pull request merged**